app/src/main/java/com/example/weather/CityPickerViewModel.kt

```diff
@@ -1,11 +1,44 @@
 package com.example.weather
 
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.weather.network.City
+import com.example.weather.network.WeatherApi
+import com.example.weather.network.WeatherApiResponse
+import kotlinx.coroutines.launch
+
+const val cityIds = "4772354,5368361,5128581,4887398,4699066,5809844,5856195,5419384,5391959,4684888,5746545,4990729,4335045"
+const val appId = "bd4472d97ca0b479dc32513cf50a1bf3"
+const val units = "metric"
 
 class CityPickerViewModel : ViewModel() {
 
-    private val _cities: List<Int> = (1..100).toList()
-    val cities: List<Int>
-    get() = _cities
+    private var _cities = MutableLiveData<List<City>>()
+
+    val cities: LiveData<List<City>> = _cities
+
+    init {
```

Comment:
let's remove this init call and move the `getCityInfo()` call to the Fragment in onViewCreated like we talked about

Reply:
In addition, we need to add some code to actually observe this cities LiveData object. Currently we are just passing this LiveData object around like any other list. We should be observing it so that we can update the data when we notice that it changes. This code will live in Fragment in the onViewCreated similar to the above comment. 
As a hint, we are going to want something like this before the call to `getCityInfo()`:
`viewModel.cities.observe() {}`

Reply:
Once we do this, we can just pass in a List<City> to the adapter instead of the LiveData<> object we are passing in now